core/state/state_object.go

```diff
@@ -220,6 +227,11 @@ func (s *stateObject) GetCommittedState(db Database, key common.Hash) common.Has
 		}()
 	}
 
+	if enc, err = s.getTrie(db).TryGet(key.Bytes()); len(enc) == 0 && err == nil {
+		// This key didnt exist in  the database, we need to add it to the uniqueKeys
+		s.uniqueNewKeysStorage[key] = common.Hash{}
```

Comment:
Since you don't use the value of uniqueNewKeysStorage, it's probably better to store struct{}{} (0 bytes) vs an empty array of 32 bytes for each key

Reply:
I am using the similar Storage type we use for the pending keys, didnt want to create my own. This wont take much memory anyways, so shouldnt be a problem i think